src/ttl_table.c

```diff
@@ -132,3 +132,170 @@ bool TimeToLiveTable_VerifyDocAndFields(TimeToLiveTable *table, t_docId docId, c
   }
   return false;
 }
+
+bool TimeToLiveTable_VerifyDocAndField(TimeToLiveTable *table, t_docId docId, t_fieldIndex field, enum FieldExpirationPredicate predicate, const struct timespec* expirationPoint) {
+  dictEntry *entry = dictFind(table, (void*)docId);
+  if (!entry) {
+    // the document did not have a ttl for itself or its fields
+    // if predicate is default then we know at least one field is valid
+    // if predicate is missing then we know the field is indeed missing since the document has no expiration for it
+    return true;
+  }
+
+  TimeToLiveEntry* ttlEntry = (TimeToLiveEntry*)dictGetVal(entry);
+  const size_t fieldWithExpirationCount = array_len(ttlEntry->fieldExpirations);
+  if (fieldWithExpirationCount == 0) {
+    // the document has no fields with expiration times, there exists at least one valid field
+    return true;
+  }
+
+  for (size_t currentFieldIndex = 0; currentFieldIndex < fieldWithExpirationCount; currentFieldIndex++) {
+    FieldExpiration* fieldExpiration = &ttlEntry->fieldExpirations[currentFieldIndex];
+    if (field == fieldExpiration->index) {
+      // the field has an expiration time
+      const bool expired = DidExpire(&fieldExpiration->point, expirationPoint);
+      if (expired) {
+        return (predicate == FIELD_EXPIRATION_MISSING);
+      } else {
+        return (predicate == FIELD_EXPIRATION_DEFAULT);
```

Comment:
I do not understand, if it did not expire, shouldn't it just return TRUE? Like it is valid, why is there any other consideration here?